### DISCUSSION TITLE
Snapcast Based Players

### DISCUSSION THREAD

**Discussion by OzGav** [music-assistant/support#1668] [Player Support]
Documentation https://music-assistant.io/player-support/snapcast/

**Comment by DerPicknicker** [comment]
Hi @OzGav and @SantiagoSotoC ,

I wrote it today also in a different thread. But I think here is the better place to discuss. 

How must the Snapcast config look if I want to control the Multiroom inside MA and MA is my only music source? 

Would you be so kind and tell more about your setup? 

Edit: Each player should be able to play different music from MA.

Thank you in advance!

**Reply by Bram-diederik** [reply]
There is no config needed. Everything works as you mention it.

**Reply by DerPicknicker** [reply]
THANK YOU THAT SOUNDS AWESOME 🤩

**Reply by SantiagoSotoC** [reply]
@DerPicknicker 
Please tell me your experience, I could not send the audio to snapcast from the home assistant addon. I found some bugs in the current beta

**Reply by DerPicknicker** [reply]
@SantiagoSotoC ..

It will take some time. I have to set up an SnapCast server as an LXC container in Proxmox first. This shouldn't take too long. The major thing is that i dont have an snapclient yet.

Are the problems on the MA side or on the snapserver side?

**Reply by SantiagoSotoC** [reply]
There are issues on the MA side, as a snapcasts client you can use an android device, web browser, or any machine running windows, macOS or linux. If you need help to run snapweb on your snapcast server, send me a message on Discord.

**Reply by DerPicknicker** [reply]
Is this the right thing https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwjs9uu37v2CAxV8wAIHHdtfCTMQFnoECBEQAQ&url=https%3A%2F%2Fgithub.com%2Fbadaix%2Fsnapweb&usg=AOvVaw2kLszO6npeUOBjTEci_0tL&opi=89978449 ? 

Edit: And if yes which branch should I use?

**Reply by SantiagoSotoC** [reply]
Yes, but don't use that repo, just install snapserver. In the configuration file you can activate snapweb. You also have to open port 1780. The configuration file is /etc/snapserver.conf

**Reply by Bram-diederik** [reply]
@DerPicknicker i have installed debian bookworm and installed the backports snapcast 0.27 version

**Reply by DerPicknicker** [reply]
@Bram-diederik do you also encountered the issues mentioned by @SantiagoSotoC ...?


@SantiagoSotoC ... I will do that. Thank you.

**Reply by Bram-diederik** [reply]
It works. 
I use debian bookworm. and 0.27 is in the testing. 

I added the backports to my debian systems.
`deb http://deb.debian.org/debian bookworm-backports main contrib non-free`

(note: this does not update your whole system only packages you mention with backport are installed)

install the packages
`apt install snapclient/bookworm-backports`
AND/OR
`apt install snapserver/bookworm-backports`

and your good to go. 

if you run raspberry pi you need to install the debian keyring using dpkg (apt will use the raspberry-pi keyring) or else apt update with the backport will fail.
download the latest all deb from http://ftp.debian.org/debian/pool/main/d/debian-archive-keyring/
`dpkg -i [downloaded file]`

**Reply by DerPicknicker** [reply]
@Bram-diederik ...

I am not using Debian. All those stuff runs on a Proxmox server as an LXC (Ubuntu) container.

Why should I use backports instead of just downloading the latest deb file and install them instead?

**Reply by DerPicknicker** [reply]
@SantiagoSotoC ...

At which point you think the snapcast issues will be resolved on the MA side? 

I got the snapcast running under alpine Linux and the used Ressources are so small which is really great. The webui is also running. I will test it with this app: https://apps.apple.com/de/app/snapcast-client/id1552559653

Later I will planning to use something like this: https://github.com/CarlosDerSeher

Could you describe what kind of bugs you already found?

**Reply by SantiagoSotoC** [reply]
@DerPicknicker In beta 78, the problem that prevented playback was fixed 
There are some problems with volume control, when the speakers are synchronized, which I hope to fix soon. Another one where the playback status was not updated in HA. For the last one I already have a solution, it will be released in the next beta.

**Reply by DerPicknicker** [reply]
@SantiagoSotoC ..

Cool! I will update asap. If MA is my main music source it shouldn't be a problem to route the audio to different snapcast devices without a single change inside the snapcast config? 

Do you know if TTS is also supported? This would be epic if you can use the snapcast devices also as an notification device.

**Reply by OzGav** [reply]
Full TTS support is yet to be fully implemented in MA. It is coming soon.

**Reply by SantiagoSotoC** [reply]
@DerPicknicker 

-  Yes, you should have no problems. Just make sure you have the necessary ports open. 
- You can leave the default, which uses  a named pipe. 
- TTS is not supported in the current state of MA, it is one of the pending things before releasing the stable version, But when it is supported in MA this provider should have no problems in any case, I will fix it when the time comes.

Any problem let me know and I will try to solve it or give you some help.

**Reply by DerPicknicker** [reply]
@SantiagoSotoC .. 

This great to know... The ports are always open due the fact it's not running in an Docker environment. It's a normal VM..

**Reply by SantiagoSotoC** [reply]
Ok take into account firewalls

**Reply by DerPicknicker** [reply]
Hi @SantiagoSotoC ... 

I tested now the player integration. But for me playback doesn't work. It seems that the MA is sending the data not in the right format. 
Here is the Log from SnapServer: 
```
2023-12-11 07-52-42.509 [Notice] (StreamServer) StreamServer::NewConnection: 192.168.178.172
2023-12-11 07-52-42.510 [Info] (Server) Hello from 13B08816-8DA5-43A1-925D-842A3BB1F9CD#0, host: localhost, v0.24.0, ClientName: Snap.Net Stream, OS: iOS 17.1.2, Arch: arm64, Protocol version: 2
2023-12-11 07-52-45.247 [Error] (StreamSessionTCP) Error reading message header of length 0: End of file
2023-12-11 07-52-45.247 [Info] (StreamServer) onDisconnect: 13B08816-8DA5-43A1-925D-842A3BB1F9CD#0
2023-12-11 07-53-22.747 [Info] (Server) Stream.AddStream("tcp://0.0.0.0:4953?name=2869a895-eff8-4380-9580-33f7b943672c")
2023-12-11 07-53-22.747 [Info] (PcmStream) PcmStream: 2869a895-eff8-4380-9580-33f7b943672c, sampleFormat: 48000:16:2
2023-12-11 07-53-22.747 [Info] (TcpStream) TcpStream host: 0.0.0.0, port: 4953, is server: 1
2023-12-11 07-53-22.748 [Error] (Server) Server::onMessageReceived exception: bind: Address in use [system:98 at /usr/include/boost/asio/detail/reactive_socket_service.hpp:161 in function 'bind'], message: {"id":174,"jsonrpc":"2.0","method":"Stream.AddStream","params":{"streamUri":"tcp://0.0.0.0:4953?name=2869a895-eff8-4380-9580-33f7b943672c"}}
2023-12-11 07-53-22.758 [Info] (Server) Stream.AddStream("tcp://0.0.0.0:4954?name=2869a895-eff8-4380-9580-33f7b943672c")
2023-12-11 07-53-22.758 [Info] (PcmStream) PcmStream: 2869a895-eff8-4380-9580-33f7b943672c, sampleFormat: 48000:16:2
2023-12-11 07-53-22.758 [Info] (TcpStream) TcpStream host: 0.0.0.0, port: 4954, is server: 1
2023-12-11 07-53-22.758 [Info] (FlacEnc) Init - compression level: 2
```

I got it working once with only one track. But not really stable. So please try to check it on your side again. I encountered also a Bug Hostname doesn't work as well. The Server Adress must be an IP-Adress. Hostname Like : SNAPSERVER doesn't work at all. I tested different Clients. SnapCastWeb-Clients, IOS Clients and only the IOS clients plays some music but only once and not really stable.

**Reply by SantiagoSotoC** [reply]
Yes,I know that an ip address is required, the use of hostname is not a priority, there are other issues with the provider as well as the snapcast server should have a static ip.
Are you sure you don't have a firewall ? Did you try with flow mode enabled ? Playback is handled by ffmpg with snapcast recommended settings.

**Reply by DerPicknicker** [reply]
No I don't use any Firewalls here. it's a normal server. The Communication (Changing Volume etc.) works almost instant but the playback is really buggy. I started a stream and it took 30s to start on the right device. The WebAPP doesn't work at all here. I can only control the volume but not stream to it. 

I can try to get it running under an Ubuntu LXC instead of using Alpine.

**Reply by SantiagoSotoC** [reply]
Try a normal vm, then go for an lxc.
Some distros have the firewall enabled by default.

**Reply by SantiagoSotoC** [reply]
Alpine lxc works in my case.
About hostnames: https://github.com/badaix/snapcast/issues/1119
It is a limitation of snapcast according to this post.

**Reply by DerPicknicker** [reply]
I changed the HOSTNAME inside the SnapCast App to the real IP address but the stream still doesn't work: 
Here is the Log from MA: 
```
2023-12-11 13:12:25.492 WARNING (MainThread) [music_assistant.streams] Received unhandled GET request to /C0DA1D53-FDB9-411C-80A3-B8FB05A774EF from 192.168.178.122
headers: <CIMultiDictProxy('User-Agent': 'Lavf/59.27.100', 'Accept': '*/*', 'Range': 'bytes=0-', 'Connection': 'close', 'Host': '192.168.178.122:8096', 'Icy-MetaData': '1')>

2023-12-11 14:22:01.647 WARNING (MainThread) [music_assistant.streams] Received unhandled GET request to /13B08816-8DA5-43A1-925D-842A3BB1F9CD from 192.168.178.122
headers: <CIMultiDictProxy('User-Agent': 'Lavf/59.27.100', 'Accept': '*/*', 'Range': 'bytes=0-', 'Connection': 'close', 'Host': '192.168.178.122:8096', 'Icy-MetaData': '1')>

2023-12-11 14:23:27.535 WARNING (MainThread) [music_assistant.streams] Received unhandled GET request to /13B08816-8DA5-43A1-925D-842A3BB1F9CD from 192.168.178.122
headers: <CIMultiDictProxy('User-Agent': 'Lavf/59.27.100', 'Accept': '*/*', 'Range': 'bytes=0-', 'Connection': 'close', 'Host': '192.168.178.122:8096', 'Icy-MetaData': '1')>

2023-12-11 14:23:31.791 WARNING (MainThread) [py.warnings] /usr/local/lib/python3.11/site-packages/music_assistant/server/providers/snapcast/__init__.py:174: RuntimeWarning: coroutine 'SnapCastProvider.cmd_volume_mute' was never awaited
  self.cmd_volume_mute(player_id, False)
```

EDIT - Some older LOGS: 
```
2023-12-11 13:09:37.403 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Snapweb client
2023-12-11 13:09:37.582 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-2051' coro=<FFmpeg.execute() done, defined at /usr/local/lib/python3.11/site-packages/ffmpeg/asyncio/ffmpeg.py:94> exception=FFmpegError('Non-zero exit status 1')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/ffmpeg/asyncio/ffmpeg.py", line 143, in execute
    raise FFmpegError(f"Non-zero exit status {self._process.returncode}")
ffmpeg.ffmpeg.FFmpegError: Non-zero exit status 1
2023-12-11 13:10:03.082 ERROR (MainThread) [music_assistant.webserver] [140070582913488] Error handling message: CommandMessage(message_id=76, command='config/players/get', args={'player_id': 'C0DA1D53-FDB9-411C-80A3-B8FB05A774EF'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 372, in _run_handler
    result = await result
             ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/config.py", line 343, in get_player_config
    raise KeyError(f"No config found for player id {player_id}")
KeyError: 'No config found for player id C0DA1D53-FDB9-411C-80A3-B8FB05A774EF'
2023-12-11 13:10:32.125 WARNING (MainThread) [music_assistant.streams] Received unhandled GET request to /C0DA1D53-FDB9-411C-80A3-B8FB05A774EF from 192.168.178.122
headers: <CIMultiDictProxy('User-Agent': 'Lavf/59.27.100', 'Accept': '*/*', 'Range': 'bytes=0-', 'Connection': 'close', 'Host': '192.168.178.122:8096', 'Icy-MetaData': '1')>

2023-12-11 13:11:13.939 WARNING (MainThread) [music_assistant.streams] Received unhandled GET request to /C0DA1D53-FDB9-411C-80A3-B8FB05A774EF from 192.168.178.122
headers: <CIMultiDictProxy('User-Agent': 'Lavf/59.27.100', 'Accept': '*/*', 'Range': 'bytes=0-', 'Connection': 'close', 'Host': '192.168.178.122:8096', 'Icy-MetaData': '1')>

2023-12-11 13:11:42.249 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Snapweb client
2023-12-11 13:12:01.429 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-2325' coro=<FFmpeg.execute() done, defined at /usr/local/lib/python3.11/site-packages/ffmpeg/asyncio/ffmpeg.py:94> exception=FFmpegError('Non-zero exit status 1')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/ffmpeg/asyncio/ffmpeg.py", line 143, in execute
    raise FFmpegError(f"Non-zero exit status {self._process.returncode}")
ffmpeg.ffmpeg.FFmpegError: Non-zero exit status 1
2023-12-11 13:12:15.070 ERROR (MainThread) [music_assistant.webserver] [140070582913488] Error handling message: CommandMessage(message_id=112, command='config/players/get', args={'player_id': 'C0DA1D53-FDB9-411C-80A3-B8FB05A774EF'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 372, in _run_handler
    result = await result
             ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/config.py", line 343, in get_player_config
    raise KeyError(f"No config found for player id {player_id}")
KeyError: 'No config found for player id C0DA1D53-FDB9-411C-80A3-B8FB05A774EF'
```

**Reply by SantiagoSotoC** [reply]
Are you using the latest beta ?
What is snapcast app?
Do you have an account on Discord ?

**Reply by DerPicknicker** [reply]
I use the latest beta I checked it. I don't have a discord account yet. 

What kind of information do you need from my side to make debugging a little bit easier?

He is the link to the app https://apps.apple.com/de/app/snapcast-client/id1552559653

**Reply by OzGav** [reply]
Please type the version number for the beta you are running as we often find people are not running the latest

**Reply by DerPicknicker** [reply]
This is the version: **2.0.0b78** which tells me Home Assistant. It looks like it's the latest version because it's mentioned the changes on the snapcast player provider.

**Reply by SantiagoSotoC** [reply]
Create an account in discord so we can solve this faster, a diagram of your network would be very useful and a copy of your snapserver configuration file.

**Reply by OzGav** [reply]
I wonder if the ios app is the problem. I just spun up snapserver on a RPi 1 and snapweb is enabled by default and it worked first go (and that never happens!)

**Reply by DerPicknicker** [reply]
@OzGav ... The snapweb does also not stream any content (Safari macOS) 

@SantiagoSotoC I will see what I can do. The config is 100% DEFAULT just the snap web interface enabled. How should I contact you on discord?

**Reply by SantiagoSotoC** [reply]
@DerPicknicker says it happens to all his clients

**Reply by SantiagoSotoC** [reply]
@DerPicknicker Subscribe to the music assistant server, write in the snapcastm channel and I will write to you.

**Reply by DerPicknicker** [reply]
Hello together, 

issues can be marked as resolved. The Player provider works fine with the latest beta. The issue is located on the IOS app. 

Thank you for the support @SantiagoSotoC . It helped me a lot. And for everyone who is interested in Multiroom audio. The sync is mind blowing. You can't hear any difference.

**Comment by Bram-diederik** [comment]
Well I got the integration working but i am not happy. 
This setup is excellent when music assistant is the main music control. 
but at my system snapcast and home assistant are. 
I will have issues integrating this method to my system.

Can you guys make it possible to limit the integration into one or more streams with a set name to snapcast. 
And play music to those streams instead directly to a speaker. 

I will have a bad time rewriting my asterisk extention generation to work with music assistant.

**Comment by Bram-diederik** [comment]
well ubuntu runs on testing so should have 0.27 my default.

Op do 7 dec 2023 om 20:52 schreef DerPicknicker ***@***.***>:

**Comment by DutchJaFO** [comment]
A few clarifications in the instructions would help me understand the setup better.

First impressions / questions :
1.  install snapcast server on a linux host
No further tweaking needed here ?
Is there a docker image that I can use instead ? (or is the snapcast server version a limiting factor here ?)
I suppose I could add a linux server to my proxmox, so this may be an excuse to test try this. 😄 

(2) install snapclient
Can I skip this if I want to use the android client ?

(3) Enable SnapWeb
Is this needed and do I do this on the snapcast server ?
Maybe swap this chapter with the snap client instructions and add an explanation for this.

**Reply by OzGav** [reply]
Yes this is all good. I’ll make some changes in the next couple of days. But to answer your questions quickly now.
1) Yes you just need to install it on some Linux host. No tweaking required. I installed it on a RPi 1 and it worked first go! If you want more info check the actual repo https://github.com/badaix/snapcast#installation

2) yes

3) No not needed if you don’t want to stream to web browsers. Yes a setting is required on the server although I think it was enabled by default when I installed it.

**Reply by DutchJaFO** [reply]
Because this might be useful to others I'm documenting my install process.

Step 1 : add a Linux LCX container to my Proxmox server 
I followed this guide to install a Debian (Bookworm / v12) as a LCX container :
https://computingforgeeks.com/run-debian-ubuntu-in-lxc-container-on-promox/

Step 2: install SnapServer
= open console on server

= download the snapcast server package with this command :
`wget https://github.com/badaix/snapcast/releases/download/v0.27.0/snapserver_0.27.0-1_amd64.deb`

= Install snap
`sudo apt install ./snapserver_0.27.0-1_amd64.deb`

Here I get an error telling me I've got unmet dependencies and I should fix them.
Initially it tells me to "fix broken install", but that appears to remove the snapserver ?
`apt --fix-broken install`

Then the advise appears to be :
`apt-get install avahi-daemon`

This appears to partially resolve things, but after yet anoter attempt at installing the snapserver I get the error that libflac8 is missing.
So I try :
`apt-get install libflac8`

Only to get the answer "Package 'libflac8' has no installation candidate"

**Reply by DutchJaFO** [reply]
Attempt 2 : using default options in Debian ... not a success. Same Libflac8 error.

**Reply by DutchJaFO** [reply]
Attempt 3 : selecting non default and picking Debian 11 (Bullseye) instead ...
Looks like this may be the ticket.

I think I need check where I'm downloading and running the apt-install step, because I was logged in as 'root' and in the 'root' directory.

**Reply by DutchJaFO** [reply]
yep ... it is working.
For those following along :
In step 2 (where you download the app package ) just make sure you're not in the 'root' directory.
I've changed directory to 'home' and that appeared to do the trick.

**Reply by DutchJaFO** [reply]
Final steps :


Step 1 : add a Linux LCX container to my Proxmox server 
I followed this guide to install a Debian (Bullseye / v11) as a LCX container :
https://computingforgeeks.com/run-debian-ubuntu-in-lxc-container-on-promox/
This does require the non-default install. Where it asks for priveleged vs unpriveleged install I choose the latter.
**Bookworm (v12) does not work as there appears to be a missing package that the snapcast server is using!**

Step 2: install SnapServer
= open console on server 

= change directory to 'home' 
The default directory 'root' causes issues in next part !

= download the snapcast server package with this command :
`wget https://github.com/badaix/snapcast/releases/download/v0.27.0/snapserver_0.27.0-1_amd64.deb`

= Install snap
`sudo apt install ./snapserver_0.27.0-1_amd64.deb`

= make a note of the server ip because you'll need it when connecting snapcast clients !

Step 3 : download and install the android snapcast client
= use the server Ip from previous step to connect to the server.

**Reply by Bram-diederik** [reply]
Why dont use the debian backport i posted?

**Reply by DutchJaFO** [reply]
because I didn't see that part of the discussion and I don't know what a backport is.

Heck ... I had to use google to find a way to create a linux virtual machine on my proxmox server.
Still not sure if it is the best way to do so, but at least it was convenient with minimal thinking required while also giving me options if I need yet another linux virtual machine in the future.

**Comment by Pittermaennchen** [comment]
I would like to use an old android smartphone (connected with a Bluetooth speaker) as snapcast client.

**But** when I search for Snapdroid in Google's Play Store no app with that name shows up. :(

**Reply by agdamsbo** [reply]
Search for Snapcast on f-droid or follow this link: https://f-droid.org/packages/de.badaix.snapcast/

**Comment by Kudhoh** [comment]
Hi,
Is the snapcast server supposed to be shipped with the snapweb interface? When i go to ip:1780, i only see the snapcast default page :
![image](https://github.com/music-assistant/hass-music-assistant/assets/31036126/2dd4efc8-9757-4c34-97ac-5c1817eeab7e)
I can interact with the server with snapdroid though.

Also, is it possible to define several streams like a snapcast server? (lbrespot, airplay...) I guess this is not supported as MA is supposed to interact directly with this server.

**Reply by OzGav** [reply]
Yes Snapweb is supposed to work so this will get looked at. Not sure what you mean by your second point? All Snapclients appear as individual players and can have their own queue.

**Reply by Bram-diederik** [reply]
De deb isnt shipped with the client i tought. You need to copy past the webclient into the webserver path.

**Reply by OzGav** [reply]
@Kudhoh This is fixed in b118

**Reply by Kudhoh** [reply]
Thanks, great work. My second point was about the possibility to use this server like a true snapserver and change its config to add streams 
`[stream]
librespot:///<path/to/librespot>?name=<name>[&dryout_ms=2000][&username=<my username>&password=<my password>][&devicename=Snapcast][&bitrate=320][&wd_timeout=7800][&volume=100][&onevent=""][&normalize=false][&autoplay=false][&cache=""][&disable_audio_cache=false][&killall=false][&params=extra-params]`

**Reply by OzGav** [reply]
I’m not understanding the use case, if you want to stream something we can do that through MA?

**Reply by Kudhoh** [reply]
A general use case I see is deploying a snapcast server on HA without the need of another Pi, for people not liking MA's interface. Not that I don't like it really, but I prefer using Spotify app. Just a preference.
The personnal use case I see is my gf uses Apple Music which is not available on MA at the moment (and I didn't give her permission to my HA instance anyway). That could go Apple App -> Snapserver (with shairport) -> Snapclient. I would switch stream to librespot or MA when I need.
Just a thought I had though, might be irrelevant.

**Reply by marcelveldt** [reply]
> A general use case I see is deploying a snapcast server on HA without the need of another Pi, for people not liking MA's interface. Not that I don't like it really, but I prefer using Spotify app. Just a preference.
> 
> The personnal use case I see is my gf uses Apple Music which is not available on MA at the moment (and I didn't give her permission to my HA instance anyway). That could go Apple App -> Snapserver (with shairport) -> Snapclient. I would switch stream to librespot or MA when I need.
> 
> Just a thought I had though, might be irrelevant.

That functionality is coming to MA itself even.

**Reply by LeoSum8** [reply]
> That functionality is coming to MA itself even.

Does that mean you are basically planning to spin up shairport or librespot instances from within MA and pipe their output through to connected players, all controlled via MA's webinterface?

**Comment by marc4s** [comment]
I had read in the changelog that snapcast is already integrated in MA. how to use it? especially the snapweb?

**Reply by DerPicknicker** [reply]
Install the latest image and enable the player provider. That's it.

The ports are the default ones (see the snapcast and snapweb GitHub for the ports)

**Reply by DutchJaFO** [reply]
that's odd ...
I'm running HA on 192.168.0.5 so I should be getting it on http://192.168.0.5:1780/  correct ?
So why is the (internal) snapcast client appearing on http://192.168.0.6:1780/ ??
I've definitely disabled/stopped the one on my system.

It would be nice to have a link to the snapcast webclient in the provider page.

**Reply by DerPicknicker** [reply]
Those ports are for snapweb yes - 1704/1705 are for the playback control.

**Reply by DutchJaFO** [reply]
looks like I've found the culprit on my system ... 
I've managed to install snapcast onto proxmox itself, which is running at the 0.6 endpoint.
Now to remove it without killing my own system.

**Comment by marc4s** [comment]
ok now I got it with the :1780 

per default not usable with reverseproxy.
is it "safe" to open the 1780 via reverseproxy (basic-auth) and do i need  a special config to get this working via reverseproxy or is it lan/vpn only?
Or do i get a lot of people connecting it ;D

**Reply by marcelveldt** [reply]
snapcast is local lan only, not reverse proxy

**Reply by DerPicknicker** [reply]
DONT OPEN PRIVATE IP:PORTS .. It’s meant to be used only in a LOCAL NETWORK…

**Comment by riddley** [comment]
I'm too stupid to pick a player apparently. I click the "No player selected" thing and I see my snapcast player in the choices and I click on it and it opens the thing with the power icon, volume control and link-to-other-players control, but.... then what? If I click the left part of the screen it still says "No player selected." I have verified that the snapcast player works. What is supposed to happen?

Also, I've deleted some snapcast players from Home | Settings | Players but they still show up in the aforementioned list even after rebooting HA.

**Reply by OzGav** [reply]
Once you open the player view on the side just tap on one of the players to select it.  Rebooting HA won’t help. Try reloading the Snapcast provider it  rebooting MA

**Comment by LeoSum8** [comment]
In the above linked documentation you mention that pause/play doesn't work with snapcast. I noticed that too. Skipping in songs sometimes also leads to weird behavior.

To me that is a major problem as I pause/resume quite often.

Do you have a general Idea how you can tackle this issue or will this be a long term issue?

**Reply by DerPicknicker** [reply]
Totally agree pause is a must. Maybe @SantiagoSotoC  can share also his opinion. Because he is the maintainer of the provider ;-)

**Reply by SantiagoSotoC** [reply]
Marcel took care of the audio part, I don't know if I will be able to improve it, I read here that I might review it again in the future.
The audio sending is the most complex part, in my opinion.

**Reply by DerPicknicker** [reply]
Obvious ping of @marcelveldt ;-)

**Comment by eath01** [comment]
Hi,
Just wanted to thank you for your great work.
I spent quite some time trying to find the best way to set up a many (stream) to many (player) snapcast configuration. Your solution is by far the simplest to implement, most comprehensive and most flexible I found.
I do think you are doing yourselves a disservice by not highlighting how easy it is, in MA, to start multiple independent streams for individual snapcast clients. And to subsequently move clients between streams, without skipping a beat (literally).
The only comparable functionality I found required setting up multiple independent mpd or mopidy servers, each with a separate client interface.
Your solution was a game changer for me and I bet you would see greater adoption if you documentation highlighted how flexible this implementation is.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
